src/platform/intel/cavs/lib/clk.c

```diff
@@ -98,28 +100,36 @@ void platform_clock_on_wakeup(void)
 
 	if (freq_idx != get_cpu_current_freq_idx()) {
 		select_cpu_clock(freq_idx, true);
-		set_cpu_current_freq_idx(freq_idx);
 	}
 }
 
 void platform_clock_on_waiti(void)
 {
 	int freq_idx = get_cpu_current_freq_idx();
 
-	*cache_to_uncache(&active_freq_idx) = freq_idx;
-
 	if (freq_idx != CPU_LPRO_FREQ_IDX) {
 		/* LPRO requests are fast, but requests for other ROs
 		 * can take a lot of time. That's why it's better to
 		 * not release active clock just for waiti,
 		 * so they can be switched without delay on wake up.
 		 */
 		select_cpu_clock(CPU_LPRO_FREQ_IDX, false);
-		set_cpu_current_freq_idx(CPU_LPRO_FREQ_IDX);
 	}
 }
+
+#else
+static inline void set_cpu_current_freq_idx(int freq_idx, bool release_unused)
+{
+	select_cpu_clock(freq_idx, true);
+}
 #endif
 
+static int clock_platform_set_cpu_freq(int clock, int freq_idx)
```

Comment:
Needs a brief comment that this is public API and the only one supposed to save the current clock restored on wakeup.

Reply:
added